app/ts/background/metadataUtils.ts

```diff
@@ -139,25 +140,21 @@ export async function identifyAddress(ethereumClientService: EthereumClientServi
 
 export async function getAddressBookEntriesForVisualiser(ethereumClientService: EthereumClientService, visualizerResult: readonly (VisualizerResult | undefined)[], simulationState: SimulationState, userAddressBook: UserAddressBook) : Promise<AddressBookEntry[]> {
 	let addressesToFetchMetadata: bigint[] = []
-	let tokenAddresses: bigint[] = []
 
 	for (const vis of visualizerResult) {
 		if (vis === undefined) continue
 		const ethBalanceAddresses = vis.ethBalanceChanges.map((x) => x.address)
-		const from = vis.tokenResults.map((x) => x.from)
-		const to = vis.tokenResults.map((x) => x.to)
-		tokenAddresses = tokenAddresses.concat(vis.tokenResults.map((x) => x.tokenAddress))
-		addressesToFetchMetadata = addressesToFetchMetadata.concat(ethBalanceAddresses, from, to)
+		const eventArguments = vis.events.map((x) => x.type !== 'NonParsed' ? x.args : []).flat()
+		const addressesInEvents = eventArguments.filter((a): a is { typeValue: { type: 'address', value: EthereumAddress }, paramName: string } => a.typeValue.type === 'address').map((a) => a.typeValue.value)
+		addressesToFetchMetadata = addressesToFetchMetadata.concat(ethBalanceAddresses, addressesInEvents, vis.events.map((a) => a.loggersAddress))
 	}
```

Comment:
```suggestion
	for (const visualization of visualizerResult) {
		const eventArguments = visualization.events.map((x) => x.type !== 'NonParsed' ? x.args : []).flat()
		const addressesInEvents = eventArguments.filter((a): a is { typeValue: { type: 'address', value: EthereumAddress }, paramName: string } => a.typeValue.type === 'address').map((a) => a.typeValue.value)
		addressesToFetchMetadata = addressesToFetchMetadata.concat(ethBalanceAddresses, addressesInEvents, visualization.events.map((a) => a.loggersAddress))
	}
```

Reply:
hmm I am not understanding your changes. You are leaving `from`, `to` and `tokenAddress` fields out from metadata retrieval

Reply:
Bleh.  This was a mistake on my end.  I just meant to rename `vis` to `visualization` and GitHub UI did the wrong thing.